### PR TITLE
Add a new duration axis type which calculates better ticks

### PR DESCRIPTION
**Pull request by josh-williams**
This scales the ticks to be more round duration values. The formatting of the ticks is still left up to the caller though. In these screenshots below, I used humanFormat for the tick formatting:

Screen Shot 2019-05-01 at 9 34 28 AM
Screen Shot 2019-05-01 at 9 35 26 AM

### CODE REVIEW

File: src/common/utils/ticks.tsx [hunk 1]
```diff
@@ -27,12 +27,51 @@ export function getMaxTicks(size: number, dimension: number) {
...
+  60 * 60, // 1 h
+  60 * 60 * 2, // 2 h
+  60 * 60 * 4, // 4 h
+  60 * 60 * 6, // 6 h
+  60 * 60 * 8, // 8 h
+  60 * 60 * 12, // 12 h
+  ONE_DAY // 24 h
+];
+
+export function getDurationTicks(scale, maxTicks) {
```
Comment (amcdnl) [inline src/common/utils/ticks.tsx]: Can you write a unit test for this?

File: src/BarChart/BarChart.tsx [hunk 2]
```diff
@@ -81,10 +81,7 @@ export class BarChart extends React.Component<BarChartProps, {}> {
         layout
       );
     } else if (type === 'waterfall') {
-      data = buildWaterfall(
-        this.props.data as ChartShallowDataShape[],
-        layout
-      );
+      data = buildWaterfall(this.props.data as ChartShallowDataShape[], layout);
```
Comment (amcdnl) [inline src/BarChart/BarChart.tsx]: Can you add a storybook story for this? Don't worry about human formatting dependency, just something simple.

File: src/common/utils/ticks.tsx (hunk 1 above)
Comment (josh-williams) [reply src/common/utils/ticks.tsx]: done

File: src/BarChart/BarChart.tsx (hunk 2 above)
Comment (josh-williams) [reply src/BarChart/BarChart.tsx]: done

**Comment by josh-williams** [comment]
New story:
Screen Shot 2019-05-01 at 11 52 07 AM

**Pull request merged**